### DISCUSSION TITLE
Why dynamic filters don't support right outer join?

### DISCUSSION THREAD

**Discussion by SeaRise** [facebookincubator/velox#10689] [Q&A]
from https://github.com/facebookincubator/velox/blob/ad1f3932d31d0f4a9f42bdf778ec230d79e5daec/velox/exec/HashProbe.cpp#L332-L335 we can see that dynamic filters do not support right outer join. 
Why is that? Semantically, right outer join and runtime filters do not seem to conflict.

**Comment by xiaodouchen** [comment]
I have the same question. Could you help to clarify this? @mbasmanova @xiaoxmeng

**Reply by mbasmanova** [reply]
@xiaodouchen That's a good question. I agree that it seems like right join could support dynamic filter pushdown since it doesn't need probe rows that do not match. Would you like to add this support?

**Reply by xiaodouchen** [reply]
@mbasmanova Thank you for your reply. I will support it.

**Reply by xiaodouchen** [reply]
@mbasmanova I support it in https://github.com/facebookincubator/velox/pull/12057. Please help review it.
